tests/e2e/vm_creation_performance_test.go

```diff
@@ -0,0 +1,228 @@
+`, name)
+
+	vm := v1alpha2.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"vm": "linux", "service": "v1"},
+		},
+		Spec: v1alpha2.VirtualMachineSpec{
+			EnableParavirtualization: true,
+			RunPolicy:                v1alpha2.RunPolicy("AlwaysOn"),
+			OsType:                   v1alpha2.OsType("Generic"),
+			Bootloader:               v1alpha2.BootloaderType("BIOS"),
+			CPU: v1alpha2.CPUSpec{
+				Cores:        1,
+				CoreFraction: "25%",
+				ModelName:    "generic-v1",
+			},
+			Memory: v1alpha2.MemorySpec{Size: "1Gi"},
+			BlockDevices: []v1alpha2.BlockDeviceSpec{
+				{
+					Type: v1alpha2.DiskDevice,
+					VirtualMachineDisk: &v1alpha2.DiskDeviceSpec{
+						Name: vmdName.Name,
+					},
+				},
+			},
+			Provisioning: &v1alpha2.Provisioning{
+				Type:     v1alpha2.ProvisioningType("UserData"),
+				UserData: vmUserData,
+			},
+		},
+	}
+	return vm
+}
+
+func VMD(namespace, name string, cvmiName v1alpha2.ClusterVirtualMachineImage) v1alpha2.VirtualMachineDisk {
+
+	vmd := v1alpha2.VirtualMachineDisk{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha2.VirtualMachineDiskSpec{
+			DataSource: &v1alpha2.VMDDataSource{
+				Type: v1alpha2.DataSourceTypeClusterVirtualMachineImage,
+				ClusterVirtualMachineImage: &v1alpha2.DataSourceNamedRef{
+					Name: cvmiName.Name,
+				},
+			},
+			PersistentVolumeClaim: v1alpha2.VMDPersistentVolumeClaim{
+				Size: resource.NewQuantity(6*1024*1024*1024, resource.BinarySI),
+			},
+		},
+	}
+	return vmd
+}
+
+func createVM(client kubeclient.Client, vm v1alpha2.VirtualMachine, vmd v1alpha2.VirtualMachineDisk, namespace string) {
+	GinkgoHelper()
+	resVMD, err := client.VirtualMachineDisks(namespace).Create(context.TODO(), &vmd, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vmd - %s", &vmd.Name)
+	resVM, err := client.VirtualMachines(namespace).Create(context.TODO(), &vm, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vm - %s", &vm.Name)
+	fmt.Println("VM", resVM.Name, "with VMD", resVMD.Name, "created")
+}
+
+var _ = Describe("Performance test 20 vm creation", Label("performance"), Ordered, ContinueOnFailure, func() {
+	const (
+		vmName                         = "perf-test-vm"
+		vmdName                        = "perf-test-vmd"
+		cvmiName                       = "ubuntu-22.04"
+		vmCount                        = 26
+		overallTimeout                 = 12 * time.Minute
+		deleteGracePeriodSeconds int64 = 30
+	)
+
+	var cvmi *v1alpha2.ClusterVirtualMachineImage
+	var vmMap = make(map[string]string)
+
+	clientConfig := kubeclient.DefaultClientConfig(&pflag.FlagSet{})
+	client, err := kubeclient.GetClientFromClientConfig(clientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain Virtualization client: %v\n", err)
+	}
```

Comment:
make clientConfig and client a variables and initialize them in BeforeAll, so Expect can be used to check for error.